Z2A UIKit/NSCoffee_withFixes/NSCoffee/Views/OrderButtonView.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  OrderButtonView.swift
+//  NSCoffee
+//
+//  Created by Dani on 28/02/2025.
+//
+
+import UIKit
+
+final class OrderButtonView: UIView, NibLoadable {
+    @IBOutlet private weak var orderButton: UIButton!
+    @IBOutlet private weak var numberOfItemsLabel: UILabel!
+    
+    private var numberOfItems: UInt = 0 {
+        didSet {
+            /* Fix: By default the Large Content Viewer
+             (implemented in a function below) will
+             show the button's configuration. In this
+             case, the icon. It is important to show
+             the number of items in the cart too.
+             */
+            
+            orderButton.largeContentTitle = String(localized: "cartButtonTitle.\(numberOfItems)")
+        }
+    }
+    
+    var buttonPressed: (() -> Void)?
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        numberOfItems = 0
+        
+        numberOfItemsLabel.textColor = .white
+        numberOfItemsLabel.isHidden = true
+        
+        numberOfItemsLabel.layer.cornerRadius = numberOfItemsLabel.frame.height / 2
+        numberOfItemsLabel.layer.backgroundColor = UIColor.red.cgColor
+        
+        orderButton.addTarget(self, action: #selector(handleOrderButtonPressed), for: .touchUpInside)
+        
+        /* Fix: The button and the label are two accessibility
+         elements. It makes more sense to express the value in
+         the badge, as an accessibility value of the component
+         as a whole. In this case, we are making the label not
+         accessible, and configuring the button's accessibilty
+         value with the text in the label later when configuring
+         the component.
+         */
+        
+        numberOfItemsLabel.isAccessibilityElement = false
+        
+        /* Fix: Because the button doesn't have a title,
+         it also lacks of a meaningful accessibility
```

Comment:
Not 100% on the grammar here. I'd write 'lacks a' or 'lacking a' I don't think I'd write 'lacks of a' but this may be valid, I'm not certain.